### ISSUE TITLE
TypeError: Cannot read property 'javascript' of undefined

### ISSUE THREAD

**Issue by oguilleux** [webpack-contrib/terser-webpack-plugin#335]
- Operating System: OS X 10.15.7
- Node Version: 12.16.0
- NPM Version: 6.14.8
- webpack Version: 5.1.2
- terser-webpack-plugin Version: 5.0.0

### Expected Behavior

Build doesn't fail

### Actual Behavior

Build fails with :
TypeError: Cannot read property 'javascript' of undefined at /node_modules/terser-webpack-plugin/dist/index.js:366:38

### Code

My config :
```
optimization: {
  minimize: true,
    minimizer: [
      new TerserPlugin({
        terserOptions: {
          output: {
            comments: false
          },
       },
      extractComments: false
    }),
  ]
}
```

### How Do We Reproduce?

Update version of webpack & terser to latest then make build of assets of project : https://github.com/oguilleux/webpack-gulp-wordpress-starter-theme

**Comment by Jerry-zhk** [comment]
encountered the same problem, i am building my typescript babel react project.

**Comment by evilebottnawi** [comment]
@Jerry-zhk why need to spam?

**Comment by evilebottnawi** [comment]
Because `terser-webpack-plugin@5` doesn't have compatibility with webpack@4, please read readme

**Comment by oguilleux** [comment]
I'm not using webpack 4 but webpack 5.

**Comment by evilebottnawi** [comment]
@oguilleux https://github.com/oguilleux/webpack-gulp-wordpress-starter-theme/blob/master/package.json#L66 really?

**Comment by oguilleux** [comment]
Please take a look at my initial post.

You asked how do we reproduce?

I answered : Update version of webpack & terser to latest then make build of assets of project : https://github.com/oguilleux/webpack-gulp-wordpress-starter-theme

So if you update package.json to webpack 5.1.2 and terser-webpack-plugin to 5.0.0 and make the build you will reproduce the bug.

**Comment by evilebottnawi** [comment]
`webpack-stream` uses `webpack@4`, so you got this problem https://github.com/shama/webpack-stream/blob/master/package.json#L37, sorry nothing to fix here

**Comment by oguilleux** [comment]
Okay thanks for highlighting this !

**Comment by oguilleux** [comment]
Solved by updating my gulp pipe and passing webpack in webpack-steam as you can [see here](https://github.com/oguilleux/webpack-gulp-wordpress-starter-theme/commit/5af38b38a8b73836c3c4d28002cbd0c0405990aa).

Now I have this showing up : 
```
[DEP_WEBPACK_WATCH_WITHOUT_CALLBACK] DeprecationWarning: A 'callback' argument need to be provided to the 'webpack(options, callback)' function when the 'watch' option is set. There is no way to handle the 'watch' option without a callback.
```
It seems similar to [this issue](https://github.com/webpack/webpack-cli/issues/1918) though my build doesn't fail. I'll wait for your PR.

**Comment by evilebottnawi** [comment]
@oguilleux yep, we are working on it, release with fix will be today/tomorrow

**Comment by Tofandel** [comment]
Any news? Still having this issue

**Comment by evilebottnawi** [comment]
@Tofandel seriously? Please read the CHANGELOG

**Comment by woldtwerk** [comment]
This happenend to me with Webpack5.15.0 and Terser5.5.1
Problem was that I only did `yarn add -D wepack` and forgot to do `yarn add -D webpack-cli`
not necessarily related to webpack4.

**Comment by james0r** [comment]
This is by far one of the funniest GitHub issues i've seen.

**Comment by alexander-akait** [comment]
Don't know why a lot of :-1: `webpack-stream` had webpack v4 in dependencies, so when you run `npm i` you will have `webpack` v4 and v5 in `node_modules`, because we previously used `import webpack from 'webpack';` and you can have webpack v4 in `node_modules/webpack` due npm logic, we got webpack v4 but API was written for webpack v5. We rewritten our code do not use `import webpack from 'webpack';` in favor `compiler.webpack`, so nobody faced with these problems. 

I don’t know where it’s so much negative in people, even more using emoji in these case is not help somebody, if you faced with a problem please provide information, so I can help you, thanks

**Comment by tomatobybike** [comment]
If you are using webpack v5 or above you do not need to install this plugin. Webpack v5 comes with the latest terser-webpack-plugin out of the box. Using Webpack v4, you have to install terser-webpack-plugin v4.

**Comment by oguilleux** [comment]
@tomatobybike You do if you want to override [its settings](https://webpack.js.org/configuration/optimization/#optimizationminimizer).

**Comment by as-zlynn-philipps** [comment]
@alexander-akait you are getting a lot of thumbs down because you are not being pleasant in your responses. Instead of getting exasperated the first time someone asks a question and saying, "Seriously?! RTFM!!!" try exercising a little more patience and empathy. I understand you are dealing with a lot more than just this one GitHub issue, but one comment here isn't spam. I'm trying to be nothing but honest and helpful here as I've sent my fair share of online messages that I regret. ✌

**Comment by alexander-akait** [comment]
@as-zlynn-philipps I don't say any angry word, just ask a question and answer there was problem, even more I describe the problem and there it should be fixed https://github.com/webpack-contrib/terser-webpack-plugin/issues/335#issuecomment-710004332, that problem https://github.com/webpack-contrib/terser-webpack-plugin/issues/335#issuecomment-709716106 is not helpful, it was clear from the first message, sometimes I can't just take and immediately answer what the problem is, but I did it the next day
